tests/src/core/geometry/testqgspolygon.cpp

```diff
@@ -2833,6 +2833,27 @@ void TestQgsPolygon::toFromWKT()
   QVERIFY( !pl2.is3D() );
   QVERIFY( !pl2.isMeasure() );
   QCOMPARE( pl2.wkbType(), Qgis::WkbType::Polygon );
+
+  // Test WKT export with compound curve
+  QgsPolygon pl3;
+  QgsLineString *ext3 = new QgsLineString();
+  ext3->setPoints( QgsPointSequence() << QgsPoint( Qgis::WkbType::Point, 0, 0 )
+                   << QgsPoint( Qgis::WkbType::Point, 0, 10 )
+                   << QgsPoint( Qgis::WkbType::Point, 10, 10 )
+                   << QgsPoint( Qgis::WkbType::Point, 10, 0 )
+                   << QgsPoint( Qgis::WkbType::Point, 0, 0 ) );
+  pl3.setExteriorRing( ext3 );
+  QgsLineString *ring3 = new QgsLineString();
+  ring3->setPoints( QgsPointSequence() << QgsPoint( Qgis::WkbType::Point, 1, 1 )
+                    << QgsPoint( Qgis::WkbType::Point, 1, 9 )
+                    << QgsPoint( Qgis::WkbType::Point, 9, 9 )
+                    << QgsPoint( Qgis::WkbType::Point, 9, 1 )
+                    << QgsPoint( Qgis::WkbType::Point, 1, 1 ) );
+  QgsCompoundCurve *compound = new QgsCompoundCurve();
+  compound->addCurve( ring3 );
+  pl3.addInteriorRing( compound );
```

Comment:
I haven't tested, but `addInteriorRing` [uses ](https://github.com/qgis/QGIS/blob/88c4063e262a1497f927e612455efc498b531a2e/src/core/geometry/qgspolygon.cpp#L240) `curveToLine` whereas `setExteriorRing` [uses ](https://github.com/qgis/QGIS/blob/88c4063e262a1497f927e612455efc498b531a2e/src/core/geometry/qgspolygon.cpp#L273) `segmentize`

Reply:
That is irrelevant for this bug because: 
```c++
if ( ring->hasCurvedSegments()  ) ... 
```
isn't `true` so `curveToLine` is never called.

Maybe it is worth another issue though if in other situations it that produecs a different output.

The relevant fact here is that `compound` is a QgsCompoundCurve and this produces a "CompoundCurve" in the generated WKT.